agent/agent.go

```diff
@@ -227,9 +233,24 @@ func (a *Agent) run(ctx context.Context) {
 				continue
 			}
 			// get the current node description
-			newNodeDescription, err := a.nodeDescriptionWithHostname(ctx)
+			newNodeDescription, gossip, err := a.nodeDescriptionWithHostname(ctx)
 			if err != nil {
 				log.G(ctx).WithError(err).WithField("agent", a.config.Executor).Errorf("agent: updated node description unavailable")
+				continue
+			}
+
+			// if we have a valid gossip status update and it has changed since the last
+			// recording send an UpdateGossipStatus message to the manager
+			if gossip != nil && gossip.MemberCount != memberCount {
```

Comment:
My understanding was we are trying to avoid adding very specific methods to the `Executor` interface (IIRC that was a comment for SetNetworkBootstrapKeys method as well). That is the reason I used the `Describe` method for fetch the gossip status as well.

Are you ok in adding a specific method for this purpose ? In that case I also don't have to tie tie it to `nodeUpdateTicker`

Aside from the implementation, apparently a concern with this approach (having the `GossipStatus` in the `Node` object) is that it can lead to too many raft writes. Are we ok in going ahead with this ?